web/src/app/components/quick-switcher/quick-switcher.component.ts

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2019 the Octant contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+import {
+  Component,
+  OnInit,
+  OnDestroy,
+  HostListener,
+  ElementRef,
+} from '@angular/core';
+import { Router } from '@angular/router';
+import { Subject, BehaviorSubject } from 'rxjs';
+import { Navigation, NavigationChild } from '../../models/navigation';
+import { NavigationService } from '../../modules/overview/services/navigation/navigation.service';
+import { untilDestroyed } from 'ngx-take-until-destroy';
+import { distinctUntilChanged, debounceTime } from 'rxjs/operators';
+
+const emptyNavigation: Navigation = {
+  sections: [],
+  defaultPath: '',
+};
+
+interface Destination {
+  title: string;
+  path: string;
+  keywords: string[];
+}
+
+@Component({
+  selector: 'app-quick-switcher',
+  templateUrl: './quick-switcher.component.html',
+  styleUrls: ['./quick-switcher.component.scss'],
+})
+export class QuickSwitcherComponent implements OnInit, OnDestroy {
+  behavior = new BehaviorSubject<Navigation>(emptyNavigation);
+
+  navigation: Navigation = emptyNavigation;
+
+  opened = false;
+
+  destinations: Destination[];
+  filteredDestinations: Destination[];
+  currentDestination = '';
+
+  input = '';
+  inputChanged: Subject<string> = new Subject<string>();
+
+  activeIndex = 0;
+
+  constructor(
+    private navigationService: NavigationService,
+    private router: Router,
+    private el: ElementRef
+  ) {
+    // wait a bit before reacting to user input
+    this.inputChanged
+      .pipe(
+        debounceTime(150),
+        distinctUntilChanged()
+      )
+      .subscribe(f => this.updateFilteredDestinations(f));
+  }
+
+  ngOnInit() {
+    this.navigationService.current
+      .pipe(untilDestroyed(this))
+      .subscribe(navigation => {
+        this.navigation = navigation;
+        this.destinations = this.buildDestinations(navigation);
+      });
+  }
+
+  ngOnDestroy() {}
+
+  identifyNavigationItem(index: number, item: NavigationChild): string {
+    return item.title;
+  }
+
+  @HostListener('window:keyup', ['$event'])
+  keyEvent(event: KeyboardEvent) {
+    if (this.opened) {
+      return;
+    }
+    if (event.key === 'k') {
+      this.resetModal();
+      this.opened = true;
+
+      // TODO(abrand): Hack for focusing on input. Need to figure this out.
+      const el = this.el;
+      setTimeout(function() {
```

Comment:
TSLint wants this to be `setTimout(() => {`

Reply:
I think it would be great to use some kind of event instead of relying on setTimeout. I just couldn't figure it out. Would love some help here.